lib/experimental/interactivity-api/directive-processing.php

```diff
@@ -14,18 +14,24 @@
  *
  * @param array $parsed_block The parsed block.
  * @param array $source_block The source block.
- * @param array $parent_block The parent block.
+ * @param WP_Block|null $parent_block The parent block.
  *
  * @return array The parsed block.
  */
-function gutenberg_interactivity_mark_root_blocks( $parsed_block, $source_block, $parent_block ) {
-	if ( ! isset( $parent_block ) && ! WP_Directive_Processor::has_root_block() ) {
-		WP_Directive_Processor::mark_root_block( $parsed_block );
+function gutenberg_interactivity_mark_root_interactive_blocks( $parsed_block, $source_block, $parent_block ) {
+	$has_interactive_direct_parent      = isset( $parent_block ) && isset( $parent_block->block_type->supports['interactivity'] ) && $parent_block->block_type->supports['interactivity'];
+	$has_interactive_other_level_parent = WP_Directive_Processor::has_root_block();
+	if ( ! $has_interactive_direct_parent && ! $has_interactive_other_level_parent ) {
```

Comment:
In which case the parent is interactive but it was not marked as a root interactive block?

Reply:
You are right, that doesn't happen. And simplifies the function, a lot.
Nice catch, I guess I copied the previous use of `$parent_block`, while seems not to be necessary.
Updated the code in https://github.com/WordPress/gutenberg/pull/57729/commits/bff6f0798479b320236fca0cb8f999ea2a320ea6

Reply:
Ok, thanks!